scripts/benchmark.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+origin=CarperAI/trlx
+branch=main
+entity=null
+only_hash=false
+only_tiny=false
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --origin) origin="$2"; shift ;;
+        --branch) branch="$2"; shift ;;
+        --public) entity='"CarperAI"' ;;
+        --only_hash) only_hash=true ;;
+        --only_tiny) only_tiny=true ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+set -ex
+dir=`mktemp -d -p .`
+cd $dir
+trap "rm -rf ../$dir" EXIT
+
+git clone --depth 1 --single-branch -b $branch https://github.com/$origin .
+
+# temporary, adding `tags` config option to old branches
+git apply ../0001-feat-configs-add-tags-config-option.patch
+
+hash=`find . -not \( -path ./.git -prune \) -not -name "*.md" -type f -print0 | sort -z | xargs -0 sha1sum | sha1sum | cut -f1 -d" "`
+git_hash=`git log --format=%h/%s/%as -n1`
+
+if [ "$only_hash" = true ]; then
+   echo "$hash"
+   echo "$git_hash"
+   exit 0
+fi
+
+python -m venv venv
+. venv/bin/activate
+pip install torch --extra-index-url https://download.pytorch.org/whl/cu117
```

Comment:
is there some way to do this without mutating the environment? since this can break stuff you had installed depending on one version of torch.
Also would probably be good to pin a torch version and tag it? Or otherwise save the `pip freeze` output somewhere in the run's metadata.

Reply:
It also failed to install for me with 
```
  Using cached https://files.pythonhosted.org/packages/bc/c3/f068337a370801f372f2f8f6bad74a5c140f6fda3d9de154052708dd3c65/Jinja2-3.1.2-py3-none-any.whl
Collecting triton==2.0.0; platform_system == "Linux" and platform_machine == "x86_64" (from torch)
  ERROR: Could not find a version that satisfies the requirement triton==2.0.0; platform_system == "Linux" and platform_machine == "x86_64" (from torch) (from versions: 0.4.1, 0.4.2)                                                 
ERROR: No matching distribution found for triton==2.0.0; platform_system == "Linux" and platform_machine == "x86_64" (from torch)                         
```

Reply:
This setup directly reflects installation instructions from the https://github.com/CarperAI/trlx#readme. However I wouldn't mind pinning every dependency down both here and there

Also this is odd because I can install `triton==2.0.0` on `uname -i == x86_64`, moreover we probably use the same hardware here